core/src/components/modal/test/inline/modal.e2e.ts

```diff
@@ -21,4 +21,44 @@ test.describe('modal: inline', () => {
 
     expect(await page.screenshot()).toMatchSnapshot(`modal-inline-dismiss-${page.getSnapshotSettings()}.png`);
   });
+
+  test('presenting should create a single root element with the ion-page class', async ({ page, skip }, testInfo) => {
+    skip.mode('md');
+    skip.rtl();
+
+    testInfo.annotations.push({
+      type: 'issue',
+      description: 'https://github.com/ionic-team/ionic-framework/issues/26117',
+    });
+
+    await page.setContent(`
+    <ion-datetime-button datetime="datetime"></ion-datetime-button>
+
+    <ion-modal>
+      <ion-datetime id="datetime" presentation="date-time"></ion-datetime>
+    </ion-modal>
+    `);
+
+    const ionModalDidPresent = await page.spyOnEvent('ionModalDidPresent');
+    const ionModalDidDismiss = await page.spyOnEvent('ionModalDidDismiss');
+    const modal = page.locator('ion-modal');
+
+    await page.locator('#date-button').click();
+    await ionModalDidPresent.next();
+
+    // Verifies that the host element exists with the .ion-page class
+    expect(await modal.evaluate((el: HTMLIonModalElement) => el.firstElementChild!.className)).toBe('ion-page');
+
+    await modal.evaluate((el: HTMLIonModalElement) => el.dismiss());
+    await ionModalDidDismiss.next();
+
+    await page.locator('#date-button').click();
+    await ionModalDidPresent.next();
+
+    // Verifies that presenting the overlay again does not create a new host element
+    expect(await modal.evaluate((el: HTMLIonModalElement) => el.firstElementChild!.className)).toBe('ion-page');
```

Comment:
Are we able to do `/ion-page/`? If so, we should do that otherwise this check will fail if more classes are added. Same thing with the below line. (This is similar to what `toHaveClass` recommends: https://playwright.dev/docs/test-assertions#locator-assertions-to-have-class)